### PR TITLE
fix: add support for empty db + improve logging

### LINKED ISSUE #16: [BUG] blockNumber undefined when started with empty database
## Current Behavior
When the DB is empty and the data fetching is started the error below is thrown.

```
Cannot read properties of undefined (reading 'blockNumber') at getLatestEventBlockNumberFromDb 
```
  
## Expected Behavior
When the DB is empty the program will start to fetch blocks from 0.

### PR DESCRIPTION
**Pull request by Abrom8**
Fixes #16 

## Proposed Changes

  - add optional chaining operators to the blockNumber checks
  - add a missing try catch
  - improve log messages

### CODE REVIEW

File: src/utils/fetchCreateTokenEvents.ts [hunk 1]
```diff
@@ -36,53 +36,58 @@ async function findBlock(blockNumber: number): Promise<BlockI> {
...
+
+    for (const event of events) {
+      const { blockNumber, transactionHash }: { blockNumber: number; transactionHash: string } = event
+      const existingEvent = await findCreateTokenEvent(transactionHash)
+      const eventBlock = await findBlock(blockNumber)
+      if (!eventBlock && !existingEvent) {
+        cleanedEvents.push({
+          blockNumber,
+          unixTimestamp: 1234567890,
+          timestamp: getDateFromUnixTimestamp(1234567890),
```
Comment (oceanByte) [inline src/utils/fetchCreateTokenEvents.ts:58]: question(blocking): is there a way to unify `timestamp` and `unixTimestamp`?

File: src/utils/fetchCreateTokenEvents.ts [hunk 2]
```diff
@@ -36,53 +36,58 @@ async function findBlock(blockNumber: number): Promise<BlockI> {
...
+          transactionHash
+        })
+        logger.info(`Added TokenCreated event: ${transactionHash} Block: ${blockNumber}`)
+        continue
+      }
+      if (!eventBlock) {
+        continue
+      }
```
Comment (oceanByte) [inline src/utils/fetchCreateTokenEvents.ts:66]: suggestion(blocking): you could save a check if you check this before the previous if (including a `existingEvent` check). Removing one level of indentation.

File: src/utils/fetchCreateTokenEvents.ts [hunk 3]
```diff
@@ -36,53 +36,58 @@ async function findBlock(blockNumber: number): Promise<BlockI> {
...
+      const { unixTimestamp: eventBlockUnixTimestamp } = eventBlock
+      const eventBlockDate = getDateFromUnixTimestamp(eventBlockUnixTimestamp)
+
+      if (existingEvent) {
+        if (!(existingEvent.unixTimestamp === 1234567890)) {
```
Comment (oceanByte) [inline src/utils/fetchCreateTokenEvents.ts:71]: nitpick(blocking): see the previous comments about early return and check against the newly created variable. The `existingEvent` check could be dropped.

File: src/utils/fetchTransactions.ts [hunk 4]
```diff
@@ -10,7 +10,7 @@ const provider = new ethers.providers.JsonRpcProvider(providerURL)
 
 async function getLatestTxBlockNumber() {
   const txArray = await Transaction.find({}).sort('-blockNumber').limit(1).exec()
```
Comment (oceanByte) [inline src/utils/fetchTransactions.ts:12]: question(non-blocking): Could we use `findOne` with sort here? Might be a little cleaner and quicker to read.

**Review by oceanByte — CHANGES_REQUESTED** [review]

**Review by moritzkirstein — CHANGES_REQUESTED** [review]

File: src/utils/fetchCreateTokenEvents.ts (hunk 1 above)
Comment (Abrom8) [reply src/utils/fetchCreateTokenEvents.ts:58]: My thought process here is to have easy access to both formats without having to convert all the time if you work with the large amount of data. The tradeoff is one conversion during fetching and some minor space costs. 

The other idea was to keep only the converted `Date` format, but I wanted to have the original value saved as well.

**Review by oceanByte — APPROVED** [review]

**Review by moritzkirstein — APPROVED** [review]

**Pull request merged**